File: server/webapp/WEB-INF/rails.new/app/presenters/api_v1/dashboard/pipeline_group_representer.rb
##########################################################################
# Copyright 2015 ThoughtWorks, Inc.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
##########################################################################

module ApiV1
  module Dashboard

    class PipelineGroupRepresenter < ApiV1::BaseRepresenter
      alias_method :pipeline_dashboard, :represented

      link :self do |opts|
        opts[:url_builder].pipeline_group_config_list_api_url
      end

      link :doc do
        'http://api.go.cd/current/#pipeline-groups'
      end

      property :getName, as: :name
      collection :pipelines, embedded: true, exec_context: :decorator, decorator: PipelineRepresenter

      def pipelines
        pipeline_dashboard.getPipelineModels()
      end
    end
  end
end

File: api/api-config-repos-v3/src/main/java/com/thoughtworks/go/apiv3/configrepos/representers/PipelineGroupRepresenter.java
/*
 * Copyright 2019 ThoughtWorks, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.thoughtworks.go.apiv3.configrepos.representers;

import com.thoughtworks.go.api.base.OutputWriter;
import com.thoughtworks.go.config.PipelineConfigs;

public class PipelineGroupRepresenter {
    private PipelineGroupRepresenter() {
    }

    public static void toJSON(OutputWriter json, PipelineConfigs group) {
        json.add("name", group.getGroup()).
                addChildList("pipelines", (w) -> group.getPipelines().forEach(
                        (pip) -> w.addChild((gw) -> PipelineConfigRepresenter.toJSON(gw, pip)))
                );
    }
}
